src/qamqpclient.cpp

```diff
@@ -660,14 +660,20 @@ QAmqpExchange *QAmqpClient::createExchange(int channelNumber)
 QAmqpExchange *QAmqpClient::createExchange(const QString &name, int channelNumber)
 {
     Q_D(QAmqpClient);
+    if (!name.isEmpty() && d->exchanges.contains(name)
+                    && (!d->exchanges[name].isNull()))
```

Comment:
what if ```d->exchanges[name].isNull()```? This hash continues to grow even though exchanges might be deleted right?